Add custom CosmJs elements to my own chain (#998)
cosmos/sdk-tutorials: academy/4-my-own-chain/external-gui.md

├── react-dom@18.0.0
├── react-scripts@5.0.1
├── react@18.0.0
├── typescript@4.6.3
└── web-vitals@2.1.4
```

From here, it is just a matter of repeating this information in a new install command:

```sh
$ npm install react-dom@18.0.0 react-scripts@5.0.1 react@18.0.0 web-vitals@2.1.4 --save-exact
$ npm install @testing-library/jest-dom@5.16.4 @testing-library/react@13.1.1 @testing-library/user-event@13.5.0 @types/jest@27.4.1 @types/react-dom@18.0.1 @types/react@18.0.5 typescript@4.6.3 --save-exact --save-dev
```

This should update your [`package.json`](https://github.com/cosmos/academy-checkers-ui/blob/4ea0bdb/package.json#L23-L55) and lock files.

</ExpansionPanel>

To make sure that your editing of `package.json` does not reformat it with Prettier, add the ignore file `.prettierignore` with the following in it:

``` [https://github.com/cosmos/academy-checkers-ui/blob/4ea0bdb/.prettierignore#L1]
package.json
```

### Fix a compilation error

There may be a Typescript compilation error on a CosmJs generated file, so add [`"downlevelIteration": true,`](https://github.com/cosmos/academy-checkers-ui/blob/4ea0bdb/tsconfig.json#L13) to `tsconfig.json`'s `.compilerOptions`.

> Confirm that you get the React simple page by running `npm start` in your own project.

Author: ```suggestion
Confirm that you get the React simple page by running `npm start` in your own project root folder.
```

> Confirm that you get the React simple page by running `npm start` in your own project too.

Reviewer: "in your project too". 

Not very clear. Can we identify the project folder to run from? "Project root" or something else?

Author: > Can we identify the project folder to run from?

If you mean _content-wise_, I am open to suggestions. The CosmJs Checkers repo contains code for the GUI and server-side scripts and more generic code in support of both.

<ExpansionPanel title="In GameContainer.tsx">

> The use of `match` in `BrowserRouter` is not recommended:

Author: ```suggestion
The use of `match` in `BrowserRouter` no longer works:
```
@AndrewB9lab
